src/rest_framework_reactive/observer.py

```diff
@@ -125,274 +124,325 @@ def _get_logging_id(self):
+            # Evaluate the observer
+            emitted_results = self._evaluate(return_emitted=return_emitted)
+            duration = time.time() - start
+
+            # Log slow observers.
+            if duration > settings['warnings']['max_processing_time']:
+                self._warning("Slow observed viewset", duration=duration)
+
+            # Remove subscribers of really slow observers.
+            if duration > settings['errors']['max_processing_time']:
+                logger.error(
+                    "Removing subscribers to extremely slow observed viewset ({})".format(
                         self._get_logging_id()
                     ),
-                    extra=self._get_logging_extra(stopped=True),
+                    extra=self._get_logging_extra(duration=duration),
                 )
+                observer.subscribers.all().delete()
 
-                stop_observer = True
-        elif self._meta.change_detection == Options.CHANGE_DETECTION_POLL:
-            # Register poller.
-            observer.poll_interval = self._meta.poll_interval
+            return emitted_results
 
-            async_to_sync(get_channel_layer().send)(
-                CHANNEL_POLL_OBSERVER,
-                {
-                    'type': TYPE_POLL_OBSERVER,
-                    'observer': self.id,
-                    'interval': self._meta.poll_interval,
-                },
+        except:  # pylint: disable=bare-except
+            logger.exception(
+                "Error while evaluating observer ({})".format(self._get_logging_id()),
+                extra=self._get_logging_extra(),
             )
-        else:
-            raise NotImplementedError(
-                "Change detection mechanism '{}' not implemented.".format(
-                    self._meta.change_detection
-                )
+
+    def _viewset_results(self):
+        """Parse results from the viewset response."""
+        results = []
+        try:
+            response = self._viewset_method(
+                self._viewset.request, *self._request.args, **self._request.kwargs
             )
 
-        # Update last evaluation time.
-        if not stop_observer:
-            observer.last_evaluation = timezone.now()
-            observer.save()
-
-        # Log viewsets with too much output.
-        max_result_length = get_queryobserver_settings()['warnings'][
-            'max_result_length'
-        ]
-        if len(results) > max_result_length:
-            logger.warning(
-                "Observed viewset returned too many results ({})".format(
-                    self._get_logging_id()
-                ),
-                extra=self._get_logging_extra(results=len(results)),
+            if response.status_code == 200:
+                results = response.data
+
+                if not isinstance(results, list):
+                    if isinstance(results, dict):
+                        # XXX: This can incidently match if a single
```

Comment:
You can check other keys to make it less likely. I think there are always four of them - `previous`, `next`, `count` and `results` - but check in the documentation.